aiida_mlip/calculations/base.py

```diff
@@ -217,7 +217,7 @@ def prepare_for_submission(
             cmd_line["arch"] = architecture
 
         model_path = None
-        if "model" in self.inputs:
+        if "model" in self.inputs and self.inputs.model is not None:
             model_path = self.inputs.model.filepath
         else:
             if "config" in self.inputs and "model" in self.inputs.config:
```

Comment:
it's checking if we have model either in the inputs or in the config. if it's in the config we leave it in the config since the config file will be passed to the cli anyway.
if there is no model in either we want to use a default model, so it downloads the default we chose. 
but if there is no architecture either it will just use janus default.

Reply:
I understand this is confusing tbh, and I think the last step of downloading could be avoided. it comes from an initial version in which the calcjob had some defaults from everything except for the model (for some reasons that is useless to explain).
Now the defaults are handled differently than they were intially and they are just the janus defaults, so this part with the model download is kind of useless.
But the defaults need to be reorganised completely (because we said we don't want to rely on janus defaults)